feat(topics): add Topics api endpoints

wanlingt opened the pull request:
## Problem

Topics do not have API endpoints

Closes #398 

## Solution

Add the following API endpoints:
```
GET /:agencyId
POST /
DELETE /:id
PUT /:id
```
These allow us to
- retrieve nested list of topics belonging to an agency
- create / update / delete a topic - for users belonging to the same agency as the topic

Additionally, `verifyUserCanModifyTopic` and `verifyUserInAgency` were added to AuthService.

## Tests

Run the unit tests for topics and agency

wanlingt committed:
feat(topics): add API endpoints for topics

permissions has not been included yet

re #398

wanlingt committed:
feat(topics): add permissions to topics controller

re #398

wanlingt committed:
test(post, auth): modify post and auth tests to include new auth functions

2 auth functions were added: verifyUserCanModifyTopic and verifyUserInAgency. They were referenced
in post and auth tests

re #398

wanlingt committed:
fix(topics): modify parameter for listTopicsUsedByAgency

re #398

wanlingt committed:
test(topics): add unit tests

re #398

LoneRifle commented on server/src/modules/topics/topics.routes.ts:
```diff
@@ -0,0 +1,80 @@
+   * @returns 400 with database error
+   */
+  router.get(
+    '/:agencyId',
```
```suggestion
    '/agency/:agencyId',
```
To line up with the jsdoc

LoneRifle replied:
A clearer alternative may be to have a path like `/agencies/:agencyId/topics`, but that would require you to modify the agency route, injecting it with the topics controller

wanlingt replied:
went with the latter suggestion - changed the route to `/agencies/:agencyId/topics`

LoneRifle reviewed (approved):
lgtm otherwise

wanlingt committed:
refactor(topics): change route from topics to agency

- move retrieval of agency's list of topics from topics.routes to agency.routes /n- modify unit
tests accordingly

re #398

wanlingt committed:
chore(topics): add todo for remaining unit tests

wanlingt committed:
Merge branch 'develop' into feat/topics/routes

wanlingt committed:
fix(post, auth): remove duplicate imports

remoe duplicate imports from merge conflict resolution
